src/EditorFeatures/Core/Shared/Tagging/EventSources/TaggerEventSources.ViewSpanChangedEventSource.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Tagging
+{
+    internal partial class TaggerEventSources
+    {
+        private class ViewSpanChangedEventSource : ForegroundThreadAffinitizedObject, ITaggerEventSource
+        {
+            private readonly ITextView textView;
+            private readonly TaggerDelay delay;
+
+            private Span? lastSpan;
+
+            public ViewSpanChangedEventSource(ITextView textView, TaggerDelay delay)
+            {
+                this.AssertIsForeground();
+                Debug.Assert(textView != null);
+                this.textView = textView;
+                this.delay = delay;
+            }
+
+            public event EventHandler<TaggerEventArgs> Changed;
+
+#pragma warning disable CS0067
+            public event EventHandler UIUpdatesPaused;
+            public event EventHandler UIUpdatesResumed;
+#pragma warning restore
+
+            public void Connect()
+            {
+                this.AssertIsForeground();
+                textView.LayoutChanged += OnLayoutChanged;
```

Comment:
LayoutChanged is a *very* chatty event, is there something less chatty we can use?